Views/Metas.liquid

```diff
@@ -0,0 +1,30 @@
+﻿{% capture faviconUrl %}{% Href "~/Themes/ShowOrchard.Theme/Images/Favicons/" %}{% endcapture %}
+
+<meta name="viewport" content="width=device-width, initial-scale=1.0">
+<link rel="shortcut icon" sizes="256x256" href="{{ faviconUrl }}favicon.ico" type="image/x-icon" />
+<link rel="image/png" sizes="36x36" href="{{ faviconUrl }}android-chrome-36x36.png" />
+<link rel="image/png" sizes="48x48" href="{{ faviconUrl }}android-chrome-48x48.png" />
+<link rel="image/png" sizes="72x72" href="{{ faviconUrl }}android-chrome-72x72.png" />
+<link rel="image/png" sizes="96x96" href="{{ faviconUrl }}android-chrome-96x96.png" />
+<link rel="image/png" sizes="144x144" href="{{ faviconUrl }}android-chrome-144x144.png" />
+<link rel="image/png" sizes="192x192" href="{{ faviconUrl }}android-chrome-192x192.png" />
+<link rel="image/png" sizes="256x256" href="{{ faviconUrl }}android-chrome-256x256.png" />
+<link rel="image/png" sizes="384x384" href="{{ faviconUrl }}android-chrome-384x384.png" />
+<link rel="image/png" sizes="512x512" href="{{ faviconUrl }}android-chrome-512x512.png" />
+<link rel="apple-touch-icon" sizes="192x192" href="{{ faviconUrl }}apple-icon.png" />
+<link rel="apple-touch-icon" sizes="57x57" href="{{ faviconUrl }}apple-icon-57x57.png" />
+<link rel="apple-touch-icon" sizes="60x60" href="{{ faviconUrl }}apple-icon-60x60.png" />
+<link rel="apple-touch-icon" sizes="72x72" href="{{ faviconUrl }}apple-icon-72x72.png" />
+<link rel="apple-touch-icon" sizes="76x76" href="{{ faviconUrl }}apple-icon-76x76.png" />
+<link rel="apple-touch-icon" sizes="114x114" href="{{ faviconUrl }}apple-icon-114x114.png" />
+<link rel="apple-touch-icon" sizes="120x120" href="{{ faviconUrl }}apple-icon-120x120.png" />
+<link rel="apple-touch-icon" sizes="144x144" href="{{ faviconUrl }}apple-icon-144x144.png" />
+<link rel="apple-touch-icon" sizes="152x152" href="{{ faviconUrl }}apple-icon-152x152.png" />
+<link rel="apple-touch-icon" sizes="180x180" href="{{ faviconUrl }}apple-icon-180x180.png" />
+
+<meta name="msapplication-TileColor" content="#ffffff">
+<meta name="msapplication-TileImage" content="{{ faviconUrl }}ms-icon-144x144.png" />
+<meta name="msapplication-TileImage" content="{{ faviconUrl }}ms-icon-150x150.png" />
+<meta name="msapplication-TileImage" content="{{ faviconUrl }}ms-icon-310x310.png" />
+<meta name="msapplication-TileImage" content="{{ faviconUrl }}ms-icon-70x70.png" />
+<meta name="theme-color" content="#ffffff">
```

Comment:
Van oka, hogy nem a Liquid RegisterLink és SetMeta tageket használtad? https://github.com/Lombiq/Orchard-Liquid-Markup#changing-global-properties-of-the-html-document

Reply:
Úgy emlékszem ott nem tudtam megadni a méretét, illetve néztem hasonlót, és ott is így csinálták (pl. Try Orchard).

Reply:
Nem biztos, hogy máshol jól csináltuk, próbáld ki légyszi, elvileg mindkettőnek jól működnie kell.

Reply:
Két dolog: Az első, hogy még nem jöttem rá, hogy a RegisterLink-nél hogyan tudok egy változóhoz hozzáfűzni egy stringet.

Reply:
Stringet összefűzni pl. így tudsz: https://github.com/Lombiq/Orchard-Liquid-Markup#helpers

Milyen sizes paraméterre gondolsz? Nem tudok róla és nem is talltam, hogy egy meta tagnek lehetne sizes nevű attribútuma. De ha olyan attributútumot akarsz használni, amit nem ad alapból a SetMeta Liquid tag, akkor hozzá is adhatsz ilyen dinamikus paraméter feature-t a Liquid modulunkhoz: a Orchard.UI.Resources.MetaEntry-ben van SetAttribute(). Ezt lehet használni a Lombiq.LiquidMarkup.Services.Tags.SetMetaTag implementációjában: ott most csak kimentjük fieldekben a paraméterket, de igazából az összes paramétert is kimenthetnénk az Initialize()-ban és használhatjuk a Render()-ben a SetAttribute()-ot.

Reply:
Nem a változóban szeretnék stringet összefűzni, hanem a RegisterLink Href attribútumánál. Ott próbálgattam, de még nem jöttem rá. Vagy kell egyáltalán? Nem lehet mindenhova egyben kiírni a path-t?

Reply:
A fent linkelt példa megmutatja, hogy hogyan tudsz válozóban stringet összefűzni, amit aztán a Hrefben használsz. Ennél közvetlenebbül nem lehet.

Reply:
Csak mivel eddig az alap favicon path egy külön változóban volt, ezért csak a különböző faviconok neveit kellett hozzáírni. De akkor ezt így itt nem tudom? Maradhat egyben minden egyes faviconnál?

Reply:
Hát pontosan így, ahogy eddig volt, nem tudod, csak a fentiekben leírtak szerint. Csináld belátásod szerint, aztán majd megnézzük, ezen légyszi tovább ne álljunk.